### ISSUE TITLE
Syslog Errors: Apparmor Denied sigar_port

### ISSUE THREAD

**Issue by rudijs** [couchbase/docker#10]
Hi,

I'm running a single instance of couchbase community 3.0.1 on Ubuntu 14.04 (virtualbox) with the official docker image.

Seems to work OK and I have a custom bucket running on port 11212.

I am seeing non stop syslog messages though:

    Aug 20 08:43:07 vbox kernel: [ 1329.103657] type=1400 audit(1440060187.486:16258): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=11461 comm="sigar_port" requested_mask="trace" denied_mask="trace" peer="docker-default"

My docker run command is:

    sudo docker run --rm --name couchbase -p 8091:8091 -p 11210:11210 -p 11212:11212 couchbase:community-3.0.1

Web admin works OK, my client code can connect and read/write OK (port 11212).

The syslog message are on the host machine.

Should I be using a different run command?

Thanks.

**Comment by rudijs** [comment]
Adding `--security-opt apparmor:unconfined` which I think disables apparmor quiets the syslog messages.

I'm not sure if this is the ideal approach to fix this though.

Here's the run command in full:

    sudo docker run --rm --name couchbase -p 8091:8091 -p 11210:11210 -p 11212:11212 --security-opt apparmor:unconfined couchbase:community-3.0.1

**Comment by ceejatec** [comment]
Yes, I was about to point you at https://github.com/SumoLogic/sumologic-collector-docker/issues/6 and https://github.com/docker/docker/issues/7276 , but looks like you got there first.

I'm not precisely sure why sigar_port uses ptrace, but that does appear to be the case. And as Docker issue 7276 points out, ptrace is explicitly disable in the enforced default AppArmor profile for Docker. So the only workaround is to use a different AppArmor profile, or use --privileged to bypass it altogether.

What I've done for our internal build Docker containers is create a custom AppArmor profile that is just like docker-default, only with ptrace enabled. These are the steps I wrote down that I did:

# 1. Copy /etc/apparmor.d/docker to /etc/apparmor.d/docker-ptrace
# 2. Edit this file to change the profile name to "docker-default-ptrace" and
#    add a line containing "ptrace," (with the trailing comma) inside the block
# 3. Run /etc/init.d/apparmor reload

Then, I start up the containers with --security-opt=apparmor:docker-default-ptrace .

It's not a pretty workaround, but it gets the job done. (The simpler solution of modifying the "docker-default" profile directly seems not to work, as Docker recreates it!)

**Comment by ceejatec** [comment]
I don't believe this is actually a Couchbase Docker image bug, so I'd like to close it if you agree.

**Comment by rudijs** [comment]
The `--privileged` flag also quiets the syslog messages:

    sudo docker run --rm --name couchbase -p 8091:8091 -p 11210:11210 -p 11212:11212 --privileged  couchbase:community-3.0.1

**Comment by rudijs** [comment]
So we have a couple options.

Sure we can close this issue.

Thanks for the prompt reply!

**Comment by padcom** [comment]
Guys, you do realize that the case, broadly and openly discussed on the internet isn't solved as of today? Using Ubuntu 14.04 and _latest_ docker (Docker version 1.9.1, build a34a1d5) the syslog is still flooded with 

```
Jan 13 02:36:45 PL-KRA-MOB-109L kernel: [ 1322.234971] audit: type=1400 audit(1452649005.787:4843): apparmor="DENIED" operation="ptrace" profile="docker-default" pid=3984 comm="oracle" requested_mask="trace" denied_mask="trace" peer="docker-default"
```

It's silly that you'd close this issue without actually doing anything about it. It should work _properly_ out of the box and not require the
